### PR TITLE
Add UnitPassingCriteria to ephys.py and shell.py

### LINKED ISSUE #229: Table to filter units based on defined selection criteria
There's already an existing function to check this on a per unit level. Need to turn this into a computed table

https://github.com/vathes/map-ephys/blob/a94289fd82625f1d6136fc50b7046864d23a240a/pipeline/ephys.py#L664

Let's make this table in the `ephys` schema, named `CriteriaPassingUnit` (I don't like this name too much, please suggest for better names)

### PR DESCRIPTION
**Pull request by davidgodinez**
Fix #229

### CODE REVIEW

**Review by ttngu207 — CHANGES_REQUESTED** [review]
@davidgodinez, your fork seems to be behind Vathes, could you pull your fork to make sure this PR does not introduce any conflicts.

**Comment by davidgodinez** [comment]
HI @ttngu207 I did the pull to ensure my fork is up to date.

File: pipeline/ephys.py [hunk 1]
```diff
@@ -629,6 +629,21 @@ class WaveformMetric(dj.Part):
...
         """
 
+
+@schema
+class UnitPassingCriteria(dj.Computed):
+    """ This table computes whether a unit_key passes a selection criteria """
+    definition = """
+    -> Unit
+    ---
+    criteria_passed : bool      #boolean based on criteria
```
Comment (ttngu207) [inline pipeline/ephys.py:639]: the comment `boolean based on criteria` is not really helpful there (doesn't really say anything more than what can be inferred from the attribute name), can you improve this a bit to let the users know what this field is about

File: pipeline/ephys.py [hunk 2]
```diff
@@ -629,6 +629,21 @@ class WaveformMetric(dj.Part):
...
+    """
+
```
Comment (ttngu207) [inline pipeline/ephys.py:641]: Will need to modify the `key_source` here, the function `check_unit_criteria` requires many tables downstream from `Unit` to be populated, e.g. UnitStat, ClusterMetric, etc.
Please add a `key_source` that ensure those other required tables are populated before this one

File: pipeline/ephys.py (hunk 1 above)
Comment (davidgodinez) [reply pipeline/ephys.py:639]: Will do

**Pull request merged**